src/os/FileStore.cc

```diff
@@ -1244,7 +1244,8 @@ int FileStore::write_op_seq(int fd, uint64_t seq)
   int ret = TEMP_FAILURE_RETRY(::pwrite(fd, s, strlen(s), 0));
   if (ret < 0) {
     ret = -errno;
-    assert(!m_filestore_fail_eio || ret != -EIO);
```

Comment:
This assert is triggered when the conditional is not true that is: if (m_filestore_fail_eio && ret == -EIO).  You've removed that mechanism with your change.

You might want to move the kill() into __ceph_assert_info() and just use the same condition for the assert_info():

assert_info(!m_filestore_fail_eio || ret != -EIO);

So that way we still return non-EIO errors to user and EIO depending on the filestore_fail_eio config value that becomes m_filestore_fail_eio.. 

Because of the "return ret" below if assert_info() doesn't abort() or exit() then the thread will start the process of returning the error to the client, while the signal is likely handled elsewhere.